GentApp/ViewModels/CompanyViewModel.cs

```diff
@@ -76,47 +77,51 @@ public class CompanyViewModel : ViewModelBase {
 			}
 		}
 
-		public void EditCompany(string name, string address, string openingHours) {
+		public async Task EditCompany(string name, string address, string openingHours) {
 			MyCompany.Name = name;
 			MyCompany.Address = address;
 			MyCompany.OpeningHours = openingHours;
 
-			SaveCompanyCommand.Execute("Company");
+			await _companyService.Update(MyCompany);
+			RaisePropertyChanged(nameof(MyCompany));
 			_navigationService.NavigateTo(nameof(MyCompanyPage));
 		}
 
-		public void AddBranch(Branch branch) {
+		public async Task AddBranch(Branch branch) {
 			MyCompany.Branches.Add(branch);
-
-			SaveCompanyCommand.Execute("Branch");
+			await _companyService.Update(MyCompany);
 			_navigationService.NavigateTo(nameof(MyCompanyPage));
+			RaisePropertyChanged(nameof(MyCompany.Branches));
 		}
 
-		public void EditBranch(string name, string address, string openingHours, BranchType type) {
+		public async Task EditBranch(string name, string address, string openingHours, BranchType type) {
 			SelectedBranch.Name = name;
 			SelectedBranch.Address = address;
 			SelectedBranch.OpeningHours = openingHours;
 			SelectedBranch.Type = type;
 
-			MyCompany.Branches[MyCompany.Branches.FindIndex(i => i.Equals(SelectedBranch))] = SelectedBranch;
-			SaveCompanyCommand.Execute("Branch");
+			MyCompany.Branches[MyCompany.Branches.FindIndex(i => i.Id.Equals(SelectedBranch.Id))] = SelectedBranch;
+			await _branchService.Update(SelectedBranch);
 			_navigationService.NavigateTo(nameof(MyCompanyPage));
+			RaisePropertyChanged(nameof(MyCompany.Branches));
 		}
 
-		public void EditBranch(List<Event> events, List<Promotion> promotions, string name) {
+		public async Task UpdateBranch(List<Event> events, List<Promotion> promotions) {
+			string branchId = SelectedBranch.Id;
 			SelectedBranch.Events = events;
 			SelectedBranch.Promotions = promotions;
 
-			MyCompany.Branches[MyCompany.Branches.FindIndex(i => i.Equals(SelectedBranch))] = SelectedBranch;
-			SaveCompanyCommand.Execute(name);
-			_navigationService.NavigateTo(nameof(EditBranchPage));
+			MyCompany.Branches[MyCompany.Branches.FindIndex(i => i.Id.Equals(SelectedBranch.Id))] = SelectedBranch;
+			await SaveCompany();
+			SelectedBranch = MyCompany.Branches.Find(b => b.Id.Equals(branchId));
 		}
 
-		public void DeleteBranch() {
-			MyCompany.Branches.RemoveAt(MyCompany.Branches.FindIndex(i => i.Equals(SelectedBranch)));
+		public async Task DeleteBranch() {
+			MyCompany.Branches.RemoveAt(MyCompany.Branches.FindIndex(i => i.Id.Equals(SelectedBranch.Id)));
 
-			SaveCompanyCommand.Execute("Company");
+			await _branchService.Delete(SelectedBranch);
```

Comment:
wrm niet via updateCompany

Reply:
Keb alles moeten aanpassen zodawe enkele requests gebruiken. Tis redelijk ingewikkeld om een parent object up te daten en dan alle children ook up te daten in 1 keer. Ik dacht dat da vanzelf ging maar blijkbaar niet :/